action.yml

```diff
@@ -113,13 +133,27 @@ runs:
         echo "The 'acrName' argument must also be provided if the 'appSourcePath' argument is provided."
         exit 1
 
-    - name: Check for application source path or previously built image when ACR name is provided
-      if: ${{ inputs.appSourcePath == '' && inputs.acrName == '' && inputs.imageToDeploy == '' }}
+    - name: Check for application source path, a previously built image or a YAML configuration file
+      if: ${{ inputs.appSourcePath == '' && inputs.imageToDeploy == '' && inputs.yamlConfigPath == '' }}
```

Comment:
Here I assume that we dropped the `&& inputs.acrName == ''`  check because this info might be contained in the YAML. Is that right? If yes, then should we have a slightly more complex check where we're looking for `(inputs.acrName == '' && inputs.imageToDeploy == '' && inputs.yamlConfigPath != '') || the contrary`? Is there a simplification I'm missing?

Reply:
It was dropped since it's no longer needed for checking that one of our core scenarios was hit:

- `appSourcePath` -- user provides application source that they want to build OR the source that contains the Dockerfile
- `imageToDeploy` -- user provides the previously built image that they want to use for the Container App
- `yamlConfigPath` -- user provides the YAML configuration file that defines the properties of the Container App

If none of these arguments are provided, then we cannot deploy a Container App as we're missing a needed piece of information, independent of if the user provided the `acrName` argument or not. The `acrName` argument is needed when the `appSourcePath` argument is provided so we have a location to push the built image to, but if we don't have the `appSourcePath` argument, then there's no point in checking if we have the `acrName`. Hopefully that clears it up.